src/edu/csus/ecs/pc2/list/ListUtilities.java

```diff
@@ -0,0 +1,165 @@
+     * @return list of files in and under dirName
+     */
+	// TODO REFACTOR replace findAll from QuickSubmitter with this method.
+    public static List<File> findAllFiles(String dirName) {
+
+        List<File> files = new ArrayList<>();
+
+        File dir = new File(dirName);
+        File[] entries = dir.listFiles();
+
+        if (entries == null) {
+            return files;
+        }
+
+        for (File f : entries) {
+            if (f.isDirectory()) {
+                List<File> subList = findAllFiles(f.getAbsolutePath());
+                files.addAll(subList);
+            } else {
+                files.add(f);
+            }
+        }
+
+        return files;
+    }
+
+	/**
+	 * Get all CDP judge's sample submission filenames.
+	 * 
+	 * Example files under config\sumit\submissions
+	 * 
+	 * @param mycontest
+	 * @param directoryName CDP config/ dir or CDP base directory
+	 * @return list of judges sample submissions
+	 */
+	// TODO REFACTOR remove getAllCDPsubmissionFileNames from QuickSubmitter
+	public static List<File> getAllCDPsubmissionFileNames(IInternalContest mycontest, String directoryName) {
+
+		Problem[] problems = mycontest.getProblems();
+		List<File> files = new ArrayList<>();
+		
+		String configDir = directoryName + File.separator + IContestLoader.CONFIG_DIRNAME;
+		if (new File(configDir).isDirectory()) {
+		    directoryName = configDir;
+		}
+
+		for (Problem problem : problems) {
+
+			// config\sumit\submissions\accepted\ISumit.java
+			String probSubmissionDir = directoryName + File.separator + problem.getShortName() + File.separator
+					+ IContestLoader.SUBMISSIONS_DIRNAME;
+			files.addAll(findAllFiles(probSubmissionDir));
+
+		}
+
+		return files;
+	}
+
+	/**
+	 * Find files (in CDP) that contain one of the Problems in the list.
+	 * 
+	 */
+    public static List<File> filterByProblems(List<File> files, List<Problem> selectedProblemList) {
+        
+        List<File> newFileList = new ArrayList<File>();
+
+        for (File file : files) {
+            selectedProblemList.forEach((prob) -> {
+                if (file.getAbsolutePath().contains(File.separator + prob.getShortName() + File.separator)) {
+                    newFileList.add(file);
+                }
+            });
+        }
+
+        return newFileList;
+
+    }
+    
+    /**
+     * Find all accepted/Yes filenames with  "accepted" in file list. 
+     * 
+     */
+    public static List<File> filterYesJudgingType (List<File> files) {
+        
+        List<File> newFileList = new ArrayList<File>();
+
+        for (File file : files) {
+            if (file.getAbsolutePath().contains(File.separator + ClicsValidator.CLICS_CORRECT_ANSWER_MSG + File.separator)) {
+                newFileList.add(file);
+            }
+        }
+
+        return newFileList;
+    }
+    
+    public static List<File> filterByLanguages(List<File> files, IInternalContest contest, List<Language> selectedLanguageList) {
```

Comment:
This method should have header comments describing what it does and what the parameters identify.

Reply:
updated comment though, if a user does not know what a List of File is.. hard to help them.